### DISCUSSION TITLE
Adding properties to state is not reactive with vue 3

### DISCUSSION THREAD

**Discussion by realityfilter** [vuejs/pinia#1131] [Help and Questions]
### Reproduction

Take the following store. The triggering of the action login does not change the state.
No mutation is recorded in the dev tools. If you uncomment the line in the state function everything works as expected.

```ts
import { defineStore } from "pinia";

interface StateProps  {
    subscription?: {
        username: string
    }
}
export const useStore = defineStore('store', {
    state: () => ({
        // subscription: undefined
    } as StateProps),
    getters: {
        isLoggedIn: (state) => state.subscription != null
    },
    actions: {
        async login() {
            this.subscription = await new Promise((resolve) => setTimeout(() => {
                resolve({ username: "pinia" })
            }));
        }
    }
});
```

### Steps to reproduce the behavior

Use store and trigger login action. The subscription is not reactive.

### Expected behavior

"If you are using Vue 2, the data you create in state follows the same rules as the data in a Vue instance, ie the state object must be plain and you need to call Vue.set() when adding new properties to it. See also: [Vue#data](https://vuejs.org/v2/api/#data). "

When using vue3 adding new state properties should work out of the box accordingly.

### Actual behavior

Adding new properties to the state is not reactive.

**Comment by justbonds** [comment]
Would love to see an answer to this!
